src/utils/cosmosdb_model.ts

```diff
@@ -71,6 +72,7 @@ export type CosmosResource = t.TypeOf<typeof CosmosResource>;
 // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
 export const CosmosResource = t.intersection([
   BaseModel,
+  t.partial({ ttl: NonNegativeNumber }),
```

Comment:
IMHO this property should be added to the `BaseModel` type because it could be used even on creation (`TN`) and is not a property added by cosmos itself (like the other properties in `TR`).